xarray/core/dataset.py

```diff
@@ -2018,13 +2019,15 @@ def func(self, *args, **kwargs):
         return func
 
     @staticmethod
-    def _binary_op(f, reflexive=False, join='inner', fillna=False):
+    def _binary_op(f, reflexive=False, join=None, fillna=False):
         @functools.wraps(f)
         def func(self, other):
             if isinstance(other, groupby.GroupBy):
                 return NotImplemented
             if hasattr(other, 'indexes'):
-                self, other = align(self, other, join=join, copy=False)
+                # if user does not specify join, default to OPTIONS['join']
+                how_to_join = join if join is not None else OPTIONS['join']
```

Comment:
Fixed it with a compromise between yours and Vince's comments.  I can't directly copy the max_width case because join is not defined within the wrapped sub-function.